src/DotNet.Testcontainers/Containers/IDockerContainer.cs

```diff
@@ -32,6 +32,12 @@ public interface IDockerContainer : IAsyncDisposable
     [NotNull]
     string MacAddress { get; }
 
+    /// <summary>Gets the Testcontainer hostname.</summary>
+    /// <value>Returns the Docker container hostname if present or an empty string instead.</value>
+    /// <exception cref="InvalidOperationException">If container was not created.</exception>
+    [NotNull]
+    string Hostname { get; }
```

Comment:
Instead of introducing an `Endpoint` property I have moved the `Hostname` from the `HostedServiceContainer` class to this interface. I think only providing the hostname instead of an uri makes more sense since we don't know which protocol and port is useful for the user of this property.